test/e2e/apibinding/apibinding_test.go

```diff
@@ -0,0 +1,217 @@
+	require.NoError(t, err)
+
+	t.Logf("Create an APIExport for it")
+	cowboysAPIExport := &apisv1alpha1.APIExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "today-cowboys",
+		},
+		Spec: apisv1alpha1.APIExportSpec{
+			LatestResourceSchemas: []string{"today.cowboys.wildwest.dev"},
+		},
+	}
+	_, err = kcpClients.Cluster(sourceWorkspaceClusterName).ApisV1alpha1().APIExports().Create(ctx, cowboysAPIExport, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("Create an APIBinding in workspace %q that points to the today-cowboys export", targetWorkspace.Name)
+	apiBinding := &apisv1alpha1.APIBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cowboys",
+		},
+		Spec: apisv1alpha1.APIBindingSpec{
+			Reference: apisv1alpha1.ExportReference{
+				Workspace: &apisv1alpha1.WorkspaceExportReference{
+					WorkspaceName: sourceWorkspace.Name,
+					ExportName:    cowboysAPIExport.Name,
+				},
+			},
+		},
+	}
+
+	_, err = kcpClients.Cluster(targetWorkspaceClusterName).ApisV1alpha1().APIBindings().Create(ctx, apiBinding, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("Make sure %s API group does NOT show up in workspace %q group discovery", wildwest.GroupName, sourceWorkspace.Name)
+	groups, err := kcpClients.Cluster(sourceWorkspaceClusterName).Discovery().ServerGroups()
+	require.NoError(t, err, "error retrieving workspace %q group discovery", sourceWorkspace.Name)
+	require.False(t, groupExists(groups, wildwest.GroupName),
+		"should not have seen %s API group in workspace %q group discovery", wildwest.GroupName, sourceWorkspace.Name)
+
+	t.Logf("Make sure %q API group shows up in workspace %q group discovery", tenancy.GroupName, sourceWorkspace.Name)
+	err = wait.PollImmediateUntilWithContext(ctx, 100*time.Millisecond, func(c context.Context) (done bool, err error) {
+		groups, err := kcpClients.Cluster(targetWorkspaceClusterName).Discovery().ServerGroups()
+		if err != nil {
+			return false, fmt.Errorf("error retrieving source workspace group discovery: %w", err)
+		}
+		if groupExists(groups, wildwest.GroupName) {
+			return true, nil
+		}
+		return false, nil
+	})
+	require.NoError(t, err)
+
+	t.Logf("Make sure cowboys API resource shows up in workspace %q group version discovery", targetWorkspace.Name)
+	resources, err := kcpClients.Cluster(targetWorkspaceClusterName).Discovery().ServerResourcesForGroupVersion(wildwestv1alpha1.SchemeGroupVersion.String())
+	require.NoError(t, err, "error retrieving workspace %q API discovery", targetWorkspace.Name)
+	require.True(t, resourceExists(resources, "cowboys"), "workspace %q discovery is missing cowboys resource", targetWorkspace.Name)
+
+	wildwestClusterClient, err := wildwestclientset.NewClusterForConfig(cfg)
+	require.NoError(t, err)
+
+	t.Logf("Make sure we can perform CRUD operations workspace %q for the bound API", targetWorkspace.Name)
+
+	t.Logf("Make sure list shows nothing to start")
```

Comment:
(No action required) Maybe delete this block? Org and workspace isolation should allow tests to assume no interference from other actors.

Reply:
IMO it has value. This is about the low-level API semantics. It's crucial that an APIImport does not get the objects from some common logical cluster.